model/registry/service.yaml

```
@@ -20,6 +20,13 @@ groups:
         type: string
         brief: >
           The version string of the service API or implementation. The format is not defined by these conventions.
+        note: |
+          When possible, service detector component MAY use
```

```suggestion
          When possible, SDKs MAY use
```

I feel "service detector component" here is a bit strange.. maybe just using "SDK" instead is clear enough? Or is it the plan to really add this to a resource detector?

---

I switched it to just say "SDKs" here. IMO, it is kind of an implementation detail. SDKs may have a service resource detector where it could be implemented. Or it could be offered as a dedicated detector. In .NET we have a helper extension to configure the service resource which could be extended to allow auto-generation of `service.version`. I don't feel the spec needs to be really opinionated about _how_ it is implemented?

---

Agreed!